ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -532,14 +532,13 @@ func (r *RayClusterReconciler) reconcilePods(instance *rayv1alpha1.RayCluster) e
 		}
 
 		runningPods := corev1.PodList{}
-		for _, aPod := range workerPods.Items {
-			if (aPod.Status.Phase == corev1.PodRunning || aPod.Status.Phase == corev1.PodPending) && aPod.ObjectMeta.DeletionTimestamp == nil {
-				runningPods.Items = append(runningPods.Items, aPod)
+		for _, pod := range workerPods.Items {
+			// TODO (kevin85421): We also need to have a clear story of all the Pod status phases, especially for PodFailed.
+			if isPodRunningOrPendingAndNotDeleting(pod) {
+				runningPods.Items = append(runningPods.Items, pod)
 			}
 		}
-		r.updateLocalWorkersToDelete(&worker, runningPods.Items)
```

Comment:
This function will update `WorkersToDelete` by filtering out Pods that are not included in `runningPods`.